Test failures are not reported for tests with code coverage

Test runner does not return exit code >0 in case of test failures. Exit code is always `0`.

```
  Suite duration: 0.589 s, Tests: 2

  335 passing (26s)
  1 pending
  10 failing

  1) OpenShift/Component
       create
         from git repository
           happy path works:
     AssertionError: expected null to equal 'Component \'comp1\' successfully created'
        at c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:172:32
        at Generator.next (<anonymous>)
        at fulfilled (c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:9:58)

  2) OpenShift/Component
       create
         from git repository
           clones the git repo if selected:
     AssertionError: expected executeCommand to have been called exactly once with arguments git.clone, git uri
        at c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:204:37
        at Generator.next (<anonymous>)
        at fulfilled (c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:9:58)

  3) OpenShift/Component
       create
         from git repository
           shows error message when repo does not exist:
     AssertionError: expected undefined to equal 'There is no git repository at provided URL.'
        at c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:223:32
        at Generator.next (<anonymous>)
        at fulfilled (c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:9:58)

  4) OpenShift/Component
       create
         from git repository
           shows error message when invalid URL provided:
     AssertionError: expected undefined to equal 'Invalid URL provided'
        at c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:232:32
        at Generator.next (<anonymous>)
        at fulfilled (c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:9:58)

  5) OpenShift/Component
       create
         from git repository
           shows error message for empty git repository url:
     AssertionError: expected undefined to equal 'Empty Git repository URL'
        at c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:241:32
        at Generator.next (<anonymous>)
        at fulfilled (c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:9:58)

  6) OpenShift/Component
       create
         from binary file
           happy path works:
     AssertionError: expected null to equal 'Component \'comp1\' successfully created'
        at c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:283:32
        at Generator.next (<anonymous>)
        at fulfilled (c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:9:58)

  7) OpenShift/Component
       undeploy
         works from context menu:
     AssertionError: expected execute to have been called with arguments odo delete comp1 -f --app app1 --project myproject --all
odo delete comp1 -f --app app1 --project myproject odo delete comp1 -f --app app1 --project myproject --all
c:\p\vscode-openshift-tools\out\test\fixtures\components\comp1
        at c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:544:30
        at Generator.next (<anonymous>)
        at fulfilled (c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:9:58)


  8) OpenShift/Component
       undeploy
         works with no context:
     AssertionError: expected execute to have been called with arguments odo delete comp1 -f --app app1 --project myproject --all
odo delete comp1 -f --app app1 --project myproject odo delete comp1 -f --app app1 --project myproject --all
c:\p\vscode-openshift-tools\out\test\fixtures\components\comp1
        at c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:549:30
        at Generator.next (<anonymous>)
        at fulfilled (c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:9:58)

  9) OpenShift/Component
       openUrl
         request to create url for component if it does not exist and exits when not confirmed:
     AssertionError: expected stub to not have been called
        at c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:952:35
        at Generator.next (<anonymous>)
        at fulfilled (c:\p\vscode-openshift-tools\out\test\unit\openshift\component.test.js:9:58)

  10) OpenShift/URL
       open
         open url in browser:
     TypeError: Cannot read property 'state' of undefined
        at Function.<anonymous> (c:\p\vscode-openshift-tools\out\src\openshift\url.js:9:7952)
        at Generator.next (<anonymous>)
        at fulfilled (c:\p\vscode-openshift-tools\out\src\openshift\url.js:9:988)



No coverage information was collected, exit without writing coverage information

Exit code:   0
Done
```

--- a/test/unit/index.coverage.ts
+++ b/test/unit/index.coverage.ts
@@ -75,7 +75,7 @@ export function run(): any {
                 mocha.run()
                     .on('fail', () => failureCount++)
                     .on('end', () => {
-                        coverageRunner.reportCoverage()
+                        coverageRunner.reportCoverage();
                         failureCount > 0 ? e(`${failureCount} tests failed.`) : c();
                     });
             } catch (error) {

--- a/test/unit/index.coverage.ts
+++ b/test/unit/index.coverage.ts
@@ -75,11 +75,12 @@ export function run(): any {
                 mocha.run()
                     .on('fail', () => failureCount++)
                     .on('end', () => {
-                        c(coverageRunner);
+                        coverageRunner.reportCoverage()
+                        failureCount > 0 ? e(`${failureCount} tests failed.`) : c();
                     });
             } catch (error) {
                 return e(error);
             }
         });
-    }).then((coverageRunner: any) => coverageRunner.reportCoverage());
+    });
 }

